include/seqan3/io/detail/parse_condition.hpp

```diff
@@ -0,0 +1,390 @@
+    //!\todo write me
+    std::tuple<std::decay_t<condition_ts>...> data;
+
+    //!\todo write me
+    template <typename value_t, std::size_t ...I>
+    inline bool explode(value_t && val, std::index_sequence<I...>) const
+    {
+        return (... || std::invoke(std::get<I>(data), std::forward<value_t>(val)));
+    }
+
+public:
+    //!\todo write me
+    static constexpr auto msg = detail::condition_message<'|', condition_ts...>;
+
+    //!\todo write me
+    constexpr or_fn() = default;
+
+    //!\todo write me
+    constexpr explicit or_fn(condition_ts ...cond_fn) : data{cond_fn...}
+    {}
+
+    //!\todo write me
+    template <typename value_t>
+    inline bool operator()(value_t && val) const
+    {
+        return explode(std::forward<value_t>(val), std::make_index_sequence<sizeof...(condition_ts)>{});
+    }
+};
+
+//!\todo write me
+template <parse_condition_concept... condition_ts>
+//\cond
+    requires sizeof...(condition_ts) >= 2
+//\endcond
+struct and_fn
+{
+private:
+    //!\todo write me
+    std::tuple<std::decay_t<condition_ts>...> data;
+
+    //!\todo write me
+    template <typename value_t, std::size_t ...I>
+    inline bool explode(value_t && val, std::index_sequence<I...>) const
+    {
+        return (... && std::invoke(std::get<I>(data), std::forward<value_t>(val)));
+    }
+
+public:
+    //!\todo write me
+    static constexpr auto msg = detail::condition_message<'&', condition_ts...>;
+
+    //!\todo write me
+    constexpr and_fn() = default;
+
+    //!\todo write me
+    constexpr explicit and_fn(condition_ts ...cond_fn) : data{cond_fn...}
+    {}
+
+    //!\todo write me
+    template <typename value_t>
+    inline bool operator()(value_t && val) const
+    {
+        return explode(std::forward<value_t>(val), std::make_index_sequence<sizeof...(condition_ts)>{});
+    }
+};
+
+//!\todo write me
+template <parse_condition_concept condition_t>
+struct not_fn
+{
+private:
+    //!\todo write me
+    condition_t cond{};
+public:
+    //!\todo write me
+    static constexpr auto msg = constexpr_string{'!'} + condition_t::msg;
+
+    //!\todo write me
+    constexpr not_fn() = default;
+
+    //!\todo write me
+    constexpr explicit not_fn(condition_t _cond) : cond{_cond}
+    {}
+
+        //!\todo write me
+    template <typename value_t>
+    inline bool operator()(value_t && v) const
+    {
+        return !cond(std::forward<value_t>(v));
+    }
+};
+
+//!\todo write me
+template <parse_condition_concept condition_t>
+struct parse_condition
+{
+private:
+
+    //!\brief Friend declaration for other parse_condition types.
+    template <typename other_condition_type>
+    friend class parse_condition;
+
+    //!\todo write me
+    condition_t cond{};
+
+public:
+    //!\todo write me
+    static constexpr constexpr_string msg = condition_t::msg;
+
+    //!\todo write me
+    constexpr parse_condition() = default;
+
+    //!\todo write me
+    explicit constexpr parse_condition(condition_t c) : cond{c}
+    {}
```

Comment:
don't we need to define also move constructor and assignment operator after modifying this?